godotproject/enemy-Ant.gd

```diff
@@ -43,8 +42,15 @@ func _physics_process(delta):
 				direction *= -1
 
 func begin_attack():
-	attackState = STATE.detected_player_e
-	$ActionDelay.start()
+	#Detect if there is a wall in between the player 
+	var RayCastList = [$Orientation/RayCast_Top.get_collider(), $Orientation/RayCast_Middle.get_collider(), $Orientation/RayCast_Bottom.get_collider()]
+	for RC in RayCastList:
+		if is_instance_valid(RC):
+			#Player was the closest for at least one raycast
+			if RC.name == "Player":
+				attackState = STATE.detected_player_e
+				$ActionDelay.start()
+				pass
```

Comment:
Ah nice, so you just create the raycast and point it somewhere, then call the get_collider() function to see what the ray hit? Smart smart 👍